src/admin/python/itest_load.py

```diff
@@ -0,0 +1,516 @@
+from typing import Any, Dict, Generator, List, NamedTuple, Optional
+import yaml
+
+from .qserv_backoff import on_backoff, max_backoff_sec
+from .mysql_connection import mysql_connection
+from .replicationInterface import ReplicationInterface
+from .template import apply_template_cfg
+
+qserv_data_dir = "/qserv/data"
+
+chunk_info_file = "chunk_info.json"
+
+_log = logging.getLogger(__name__)
+
+
+def unzip(source: str, destination: str) -> None:
+    """Unzip a source file and write it to the target location."""
+    os.makedirs(os.path.dirname(destination), exist_ok=True)
+    with gzip.open(source, "rb") as _source, open(destination, "wb") as _target:
+        shutil.copyfileobj(_source, _target)
+
+
+@backoff.on_exception(
+    exception=mysql.connector.errors.DatabaseError,
+    wait_gen=backoff.expo,
+    on_backoff=on_backoff(log=_log),
+    max_time=max_backoff_sec,
+)
+def _create_ref_db(ref_db_admin: str, name: str) -> None:
+    """Create a database in the mysql used for integration test reference.
+
+    Parameters
+    ----------
+    ref_db_admin : `str`
+        URI to the reference db for the admin user.
+    name : `str`
+        The name of the database to create.
+    """
+    create_stmt = """CREATE USER IF NOT EXISTS '{{ mysqld_user_qserv }}'@'localhost';
+    CREATE USER IF NOT EXISTS '{{ mysqld_user_qserv }}'@'%';
+    CREATE DATABASE IF NOT EXISTS {name};
+    """
+    grant_stmt = """GRANT ALL ON {name}.* TO '{{ mysqld_user_qserv }}'@'localhost';
+    GRANT ALL ON {name}.* TO '{{ mysqld_user_qserv }}'@'%';
+    FLUSH PRIVILEGES;"""
+    for stmt in (create_stmt, grant_stmt):
+        cnx = mysql_connection(uri=ref_db_admin)
+        stmt = apply_template_cfg(stmt)
+        # `format` must be called after `apply_template_cfg`, because `format`
+        # consumes the inner braces (changes '{{ mysql_user_qserv}}' to
+        # '{mysql_user_qserv }') and so breaks the jinja templating syntax.
+        stmt = stmt.format(name=name)
+        _log.debug(f"_create_ref_db stmt:{stmt}")
+        with closing(cnx.cursor()) as cursor:
+            res = cursor.execute(stmt, multi=True)
+        cnx.close()
+
+
+def _create_ref_table(uri: str, db: str, schema_file: str) -> None:
+    """Create a table in the mysql database used for integration test reference.
+
+    Parameters
+    ----------
+    uri : `str`
+        The URI to the reference database for a user that has permission to
+        create a table.
+    db : `str`
+        The name of the database to create the table in.
+    schema_file : `str`
+        Absolute path to a file that contains the schema used to create the
+        table.
+    """
+    cnx = mysql_connection(uri)
+    _log.debug(f"_create_ref_table schema_file:{schema_file}")
+    if not cnx.is_connected():
+        cnx.connect()
+    cursor = cnx.cursor()
+    cursor.execute(f"USE {db}")
+    with open(schema_file) as f:
+        cursor.execute(f.read(), multi=True)
+    cursor.close()
+    cnx.close()
```

Comment:
For consistency I'd suggest `closing` context manager for cursor and connection.